src/nft/components/sell/modal/ListingButton.tsx

```diff
@@ -0,0 +1,256 @@
+    setCanContinue(continueCheck)
+    return [
+      noMarketplacesSelected,
+      missingExpiration,
+      invalidExpiration,
+      overMaxExpiration,
+      listingsMissingPrice,
+      listingsBelowFloor,
+      listingsAboveSellOrderFloor,
+      invalidPrices,
+    ]
+  }, [sellAssets])
+
+  const [disableListButton, warningMessage] = useMemo(() => {
+    const disableListButton =
+      noMarketplacesSelected ||
+      missingExpiration ||
+      invalidExpiration ||
+      overMaxExpiration ||
+      invalidPrices.length > 0 ||
+      listingsMissingPrice.length > 0
+
+    const warningMessage = noMarketplacesSelected
+      ? 'No marketplaces selected'
+      : missingExpiration
+      ? 'Set duration'
+      : invalidExpiration
+      ? 'Invalid duration'
+      : overMaxExpiration
+      ? 'Max duration is 6 months'
+      : listingsMissingPrice.length > 0
+      ? `${listingsMissingPrice.length} item price${pluralize(listingsMissingPrice.length)} not set`
+      : invalidPrices.length > 0
+      ? `${invalidPrices.length} price${pluralize(invalidPrices.length)} are invalid`
+      : listingsBelowFloor.length > 0
+      ? `${listingsBelowFloor.length} item${pluralize(listingsBelowFloor.length)} listed below floor`
+      : listingsAboveSellOrderFloor.length > 0
+      ? `${listingsAboveSellOrderFloor.length} item${pluralize(listingsAboveSellOrderFloor.length)} already listed`
+      : ''
+    return [disableListButton, warningMessage]
+  }, [
+    noMarketplacesSelected,
+    missingExpiration,
+    invalidExpiration,
+    overMaxExpiration,
+    listingsMissingPrice,
+    invalidPrices,
+    listingsBelowFloor,
+    listingsAboveSellOrderFloor,
+  ])
+
+  useEffect(() => {
+    setShowWarning(false)
+  }, [warningMessage])
+
+  const addWarningMessages = () => {
+    removeAllMarketplaceWarnings()
+    if (!missingExpiration && !noMarketplacesSelected) {
+      if (listingsMissingPrice.length > 0) {
+        for (const [asset, listing] of listingsMissingPrice) {
+          addMarketplaceWarning(asset, {
+            message: 'PLEASE SET A PRICE',
+            marketplace: listing.marketplace,
+          })
+        }
+      } else if (invalidPrices.length > 0) {
+        for (const [asset, listing] of invalidPrices) {
+          !listing.overrideFloorPrice &&
+            addMarketplaceWarning(asset, {
+              message: `INVALID PRICE`,
+              marketplace: listing.marketplace,
+            })
+        }
+      }
+    }
+    setShowWarning(true)
+  }
+
+  const warningWrappedClick = () => {
+    if ((!disableListButton && canContinue) || showWarningOverride) onClick()
+    else addWarningMessages()
+  }
+
+  return (
+    <Box style={{ position: 'relative' }} width="full">
+      {!showWarningOverride && showWarning && warningMessage.length > 0 && (
+        <Row
+          className={`${bodySmall} ${styles.warningTooltip}`}
```

Comment:
can we use clsx here?

Reply:
clsx is not needed here as we don't have any conditional logic for determining the classname